src/SelectField/SelectField.js

```diff
@@ -0,0 +1,335 @@
+};
+
+type AllProps = DefaultProps & Props;
+
+type State = {
+  anchorEl: Object,
+  ignoreFocusOnce: boolean,
+  open: boolean,
+  selectedIndex: number,
+};
+
+class SelectField extends Component<DefaultProps, AllProps, State> {
+  props: AllProps;
+  static defaultProps = {
+    compareFunction: (currentValue, itemValue) => currentValue === itemValue,
+    disabled: false,
+  };
+  state = {
+    anchorEl: undefined,
+    ignoreFocusOnce: false,
+    open: false,
+    selectedIndex: undefined,
+  };
+
+  handleMouseDown = event => event.preventDefault();
+
+  handleKeyDown = event => {
+    if (OPEN_MENU_KEYS.includes(keycode(event))) {
+      event.preventDefault();
+      this.setState({ open: true, anchorEl: event.currentTarget });
+    }
+  };
+
+  handleClick = event => {
+    event.currentTarget.focus();
+    this.setState({ open: true, anchorEl: event.currentTarget });
+  };
+
+  handleRequestClose = () => this.setState({ open: false });
+
+  handleSelectFocus = event => {
+    if (this.state.ignoreFocusOnce) {
+      event.stopPropagation();
+      this.setState({ ignoreFocusOnce: false });
+    }
+  };
+
+  handleSelectBlur = event => {
+    if (this.state.open) {
+      event.stopPropagation();
+    }
+  };
+
+  handleItemClick = (event, index, value) => {
+    event.persist();
+    this.setState(
+      {
+        open: false,
+        ignoreFocusOnce: true,
+      },
+      () => {
+        if (this.props.onChange) {
+          this.props.onChange(event, index, value);
+        }
+      },
+    );
+  };
+
+  render() {
+    const {
+      children,
+      autoFocus,
+      className,
+      compareFunction,
+      defaultValue,
+      disabled,
+      error,
+      hideLabel,
+      id,
+      inputClassName,
+      InputClassName,
+      inputProps: inputPropsProp,
+      InputProps,
+      inputRef,
+      label,
+      labelClassName,
+      onDirty,
+      onClean,
+      onChange, // eslint-disable-line no-unused-vars
+      menuClassName,
+      menuProps,
+      InputLabelProps,
+      helperText,
+      helperTextClassName,
+      FormHelperTextProps,
+      fullWidth,
+      required,
+      type,
+      name,
+      placeholder,
+      rootRef,
+      value,
+      ...other
+    } = this.props;
+    const initialShrink = value !== '' && typeof value !== 'undefined';
+
+    let inputProps = {
+      onClick: this.handleClick,
+      onMouseDown: this.handleMouseDown,
+      onSelectFocus: this.handleSelectFocus,
+      onSelectBlur: this.handleSelectBlur,
+      options: children,
+      label,
+      ...inputPropsProp,
+    };
+
+    if (inputClassName) {
+      inputProps = {
+        className: inputClassName,
+        ...inputProps,
+      };
+    }
+
+    return (
+      <FormControl
+        fullWidth={fullWidth}
+        ref={rootRef}
+        className={className}
+        error={error}
+        required={required}
+        {...other}
+      >
+        {label &&
+          !(hideLabel && value) &&
```

Comment:
It it correct condition? We show `InputLabel` only if `label` is present and not both `hideLabel` is true and `value` is present. So do we want to show `InputLabel` if `value` is empty but `hideLabel` is true?